File: clients/chat-room/cloud/python/producer.py
from kafka import KafkaProducer
import json
class ChatProducer:
  def __init__(self, brokers, topic):
    self.topic = topic
    self.producer = KafkaProducer(
      bootstrap_servers=brokers,
      sasl_mechanism="SCRAM-SHA-256",
      security_protocol="SASL_SSL",
      sasl_plain_username="redpanda-chat-account",
      sasl_plain_password="<password>",
      value_serializer=lambda v: json.dumps(v).encode("utf-8"),
    )
  def send_message(self, user, message):
    self.producer.send(self.topic, {"user": user, "message": message})
    self.producer.flush()
  def close(self):
    self.producer.close()

File: clients/chat-room/cloud/nodejs/src/producer.ts
import { Kafka } from "kafkajs";
const redpanda = new Kafka({
  brokers: ["<bootstrap-server-address>"],
  ssl: {
    },
  sasl: {
    mechanism: "scram-sha-256",
    username: "redpanda-chat-account",
    password: "<password>"
  }
});
const producer = redpanda.producer();
export async function getConnection(user: string) {
  try {
    await producer.connect();
    return async (message: string) => {
      await producer.send({
        topic: "chat-room",
        messages: [{ value: JSON.stringify({ message, user }) }],
      });
    };
  } catch (error) {
    console.error("Error:", error);
  }
}
export async function disconnect() {
  try {
    await producer.disconnect();
  } catch (error) {
    console.error("Error:", error);
  }
}